Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                {
+                    if (status[0] == testInstance.NativeStatus.Name)
+                    {
+                        existing = true;
+                        status[1] = (Int32.Parse(status[1]) + 1).ToString();
+                    }
+                }
+                if (!existing) { TSItem.TestSetStatuses.Add(new string[] { testInstance.NativeStatus.Name, "1" }); }
+            }
+            return TSItem;
+        }
+
+        public BusinessFlow ConvertQCTestSetToBF(QC.QCTestSet testSet)
+        {
+            try
+            {
+                if (testSet == null) return null;
+
+                TestSuite tsLatest = Task.Run(() =>
+                {
+                    LogicalQueryPhrase test_Suite = new LogicalQueryPhrase("id", testSet.TestSetID, ComparisonOperator.Equal);                    
+                    return octaneRepository.GetEntities<TestSuite>(GetLoginDTO(), new List<IQueryPhrase>() { test_Suite });
+                }).Result.FirstOrDefault();
+                //Create Business Flow
+                BusinessFlow busFlow = new BusinessFlow();
+                busFlow.Name = testSet.TestSetName;
+                busFlow.ExternalID = testSet.TestSetID;
+                busFlow.Status = BusinessFlow.eBusinessFlowStatus.Development;
+                busFlow.Description = StripHTML(tsLatest.GetStringValue("description"));                
+                busFlow.Activities = new ObservableList<Activity>();
+                busFlow.Variables = new ObservableList<VariableBase>();
+                Dictionary<string, string> busVariables = new Dictionary<string, string>();//will store linked variables
+
+                //Create Activities Group + Activities for each TC
+                foreach (QC.QCTSTest tc in testSet.Tests)
+                {
+                    //check if the TC is already exist in repository
+                    ActivitiesGroup tcActivsGroup;
+                    ActivitiesGroup repoActivsGroup = null;
+                    if (tc.LinkedTestID != null && tc.LinkedTestID != string.Empty)
+                        repoActivsGroup = GingerActivitiesGroupsRepo.Where(x => x.ExternalID == tc.LinkedTestID).FirstOrDefault();
+                    if (repoActivsGroup == null)
+                        repoActivsGroup = GingerActivitiesGroupsRepo.Where(x => x.ExternalID == tc.TestID).FirstOrDefault();
+                    if (repoActivsGroup != null)
+                    {
+                        List<Activity> repoNotExistsStepActivity = GingerActivitiesRepo.Where(z => repoActivsGroup.ActivitiesIdentifiers.Select(y => y.ActivityExternalID).ToList().Contains(z.ExternalID))
+                                                                                       .Where(x => !tc.Steps.Select(y => y.StepID).ToList().Contains(x.ExternalID)).ToList();
+
+                        tcActivsGroup = (ActivitiesGroup)((ActivitiesGroup)repoActivsGroup).CreateInstance();
```

Comment:
Codacy found an issue: [Remove this unnecessary cast to 'ActivitiesGroup'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)